### PR TITLE
Move TableDescriptionSupplier to separate package

### PR DESCRIPTION
**Pull request by elonazoulay**

### CODE REVIEW

**Comment by kokosing** [comment]
```
2020-11-22T19:58:29.7996779Z [ERROR] Failures: 
2020-11-22T19:58:29.7999868Z [ERROR]   TestKafkaConfig.testExplicitPropertyMappings:63 Unsupported properties: [kafka.table-description-supplier]
2020-11-22T19:58:29.8001846Z [ERROR]   TestKafkaIntegrationSmokeTest>AbstractTestQueryFramework.init:92->createQueryRunner:111 » ApplicationConfiguration
2020-11-22T19:58:29.8002823Z [INFO] 
2020-11-22T19:58:29.8003200Z [ERROR] Tests run: 138, Failures: 2, Errors: 0, Skipped: 122
```

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaConnectorModule.java [hunk 1]
```diff
@@ -88,4 +99,24 @@ protected Type _deserialize(String value, DeserializationContext context)
...
+    {
+        install(installModuleIf(
+                KafkaConfig.class,
+                kafkaConfig -> name.equalsIgnoreCase(kafkaConfig.getTopicSchemaProvider()),
+                module));
+    }
+
+    @Provides
+    @Singleton
+    public TopicSchemaProvider getTopicSchemaProvider(KafkaConfig config, Map<String, TopicSchemaProvider> topicSchemaProviders)
```
Comment (kokosing) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaConnectorModule.java:113]: Why do you need that? `TopicSchemaProvider ` should be provided by `FileTopicSchemaProviderModule` and any other `*TopicSchemaProviderModule`. There should be no need for map binder. 

See `io.prestosql.plugin.hive.metastore.HiveMetastoreModule`, where each modules provides different implementaiton of `HiveMetastore`.

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/MapBasedTopicSchemaProvider.java [hunk 2]
```diff
@@ -23,12 +24,12 @@
 
 import static java.util.Objects.requireNonNull;
 
-public class MapBasedTableDescriptionSupplier
-        implements TableDescriptionSupplier
+public class MapBasedTopicSchemaProvider
+        implements TopicSchemaProvider
```
Comment (kokosing) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/MapBasedTopicSchemaProvider.java:28]: Extract rename to separate commit.

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/file/FileTopicSchemaProviderModule.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import static io.airlift.configuration.ConfigBinder.configBinder;
+
+public class FileTopicSchemaProviderModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        configBinder(binder).bindConfig(FileTopicSchemaProviderConfig.class);
+        newMapBinder(binder, String.class, TopicSchemaProvider.class).addBinding(FileTopicSchemaProvider.NAME).toProvider(FileTopicSchemaProvider.class).in(Scopes.SINGLETON);
```
Comment (kokosing) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/file/FileTopicSchemaProviderModule.java:31]: this is not needed. Just bind `TopicSchemaProvider` to `FileTopicSchemaProvider`

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/KafkaQueryRunner.java [hunk 4]
```diff
@@ -160,16 +164,19 @@ private static DistributedQueryRunner createKafkaQueryRunner(
...
+                                    .addBinding(TEST)
+                                    .toInstance(new MapBasedTopicSchemaProvider(topicDescriptions)))));
             queryRunner.installPlugin(kafkaPlugin);
 
             Map<String, String> kafkaProperties = new HashMap<>(ImmutableMap.copyOf(extraKafkaProperties));
             kafkaProperties.putIfAbsent("kafka.nodes", testingKafka.getConnectString());
+            kafkaProperties.putIfAbsent("kafka.table-description-supplier", TEST);
             kafkaProperties.putIfAbsent("kafka.default-schema", "default");
             kafkaProperties.putIfAbsent("kafka.messages-per-split", "1000");
-            kafkaProperties.putIfAbsent("kafka.table-description-dir", "write-test");
```
Comment (kokosing) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/KafkaQueryRunner.java:172]: Was this needed by tests?

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/KafkaQueryRunner.java [hunk 5]
```diff
@@ -160,16 +164,19 @@ private static DistributedQueryRunner createKafkaQueryRunner(
...
                     extensions,
-                    binder -> newSetBinder(binder, TableDescriptionSupplier.class)
-                            .addBinding()
-                            .toInstance(new MapBasedTableDescriptionSupplier(topicDescriptions))));
+                    installModuleIf(
+                            KafkaConfig.class,
+                            kafkaConfig -> kafkaConfig.getTopicSchemaProvider().equalsIgnoreCase(TEST),
+                            binder -> newMapBinder(binder, String.class, TopicSchemaProvider.class)
+                                    .addBinding(TEST)
+                                    .toInstance(new MapBasedTopicSchemaProvider(topicDescriptions)))));
```
Comment (kokosing) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/KafkaQueryRunner.java:172]: I like that!

**Review by kokosing — COMMENTED** [review]
Thanks!

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/KafkaQueryRunner.java (hunk 4 above)
Comment (elonazoulay) [reply presto-kafka/src/test/java/io/prestosql/plugin/kafka/KafkaQueryRunner.java:172]: No, the tests passed without it locally.

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/schema/file/TestFileTopicSchemaProviderConfig.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        assertRecordedDefaults(recordDefaults(FileTopicSchemaProviderConfig.class)
+                .setTableDescriptionDir(new File("etc/kafka"))
+                .setTableNames(""));
+    }
+
+    @Test
+    public void testExplicitPropertyMappings()
+    {
+        Map<String, String> properties = new ImmutableMap.Builder<String, String>()
+                .put("kafka.table-description-dir", "/var/lib/kafka")
```
Comment (kokosing) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/schema/file/TestFileTopicSchemaProviderConfig.java:40]: This is strange that `TopicSchemaProviderConfig` contains `table-description-dir`. Maybe the old name of `TableDescriptionSupplier` was not that bad. WDYT?

Comment (elonazoulay) [reply presto-kafka/src/test/java/io/prestosql/plugin/kafka/schema/file/TestFileTopicSchemaProviderConfig.java:40]: Sure, it does describe the purpose of the abstraction, will revert.

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/MapBasedTopicSchemaProvider.java (hunk 2 above)
Comment (elonazoulay) [reply presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/MapBasedTopicSchemaProvider.java]: Now the name is unchanged but they were moved to a different package, should that be extracted to a separate commit?

**Review by kokosing — APPROVED** [review]

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/file/FileTableDescriptionSupplierModule.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+import com.google.inject.Binder;
+import com.google.inject.Scopes;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.prestosql.plugin.kafka.schema.TableDescriptionSupplier;
+
+import static io.airlift.configuration.ConfigBinder.configBinder;
+
+public class FileTableDescriptionSupplierModule
+        extends AbstractConfigurationAwareModule
```
Comment (Praveen2112) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/file/FileTableDescriptionSupplierModule.java:24]:
```suggestion
        implemnta Module
```
Your thoughts ?

**Review by Praveen2112 — APPROVED** [review]
Minor comments

**Comment by kokosing** [comment]
Merged as https://github.com/prestosql/presto/commit/696e10ba9e2cd292554b74f24e824719cf5a2449 (change a commit message a bit). Thanks!

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/file/FileTableDescriptionSupplierModule.java (hunk 7 above)
Comment (kokosing) [reply presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/file/FileTableDescriptionSupplierModule.java:24]: Sorry, I haven't noticed you review and merged. 

Here using `AbstractConfigurationAwareModule` is ok as it passes configuration factory to nested guice modules.
